frontend/src/advising/Templates/TemplateInfoPage.tsx

```diff
@@ -67,9 +83,150 @@ interface DropdownProps {
 
 interface NameFieldProps {
   readonly name: string;
-  readonly setTemplateName: (name: string) => void;
+  readonly setName: (name: string) => void;
+  readonly label: string;
+  readonly error: boolean;
 }
 
+interface FolderContext {
+  readonly folders: IFolderData[];
+  readonly setSelectedFolderId: (selectedFolderId: number | null) => void;
+  readonly newFolderName: string;
+  readonly setNewFolderName: (newFolderName: string) => void;
+  readonly hasDuplicateFolderName: boolean;
+}
+
+const folderContext = createContext<Partial<FolderContext>>({});
+
+const NameField: React.FC<NameFieldProps> = ({
+  name,
+  setName,
+  label,
+  error,
+}) => {
+  return (
+    <TextField
+      id="outlined-basic"
+      label={label}
+      variant="outlined"
+      value={name}
+      onChange={event => setName(event.target.value)}
+      placeholder=""
+      style={{ width: "100%" }}
+      error={error}
+    />
+  );
+};
+
+const DisplayedFolderInputField: React.FC<{ creatingNewFolder: boolean }> = ({
```

Comment:
This represents solely the conditionally displayed input field, whether that be a dropdown of existing folders or a text field for entering a new folder